include/seqan/platform.h

```diff
@@ -399,7 +399,7 @@ typedef int8_t __int8;     // nolint
 
 // A macro to eliminate warnings on GCC and Clang
 #if defined(COMPILER_GCC) || defined(COMPILER_CLANG) || defined(COMPILER_LINTEL)
-#define SEQAN_UNUSED __attribute__((unused))
+#define SEQAN_UNUSED [[gnu::unused]]
```

Comment:
can we seperate this out into an extra PR and implement also c++17's ``[[maybe_unused]]`` similar to how we are doing it with fallthrough?

Reply:
Whoa, maybe, yes. But you need to merge it asap.